lib/web_ui/lib/src/engine/text/font_collection.dart

```diff
@@ -187,19 +195,40 @@ class FontManager {
     String asset,
     Map<String, String> descriptors,
   ) {
+    Future<DomFontFace?> fontFaceLoad(DomFontFace fontFace) async {
+      try {
+        return fontFace.load();
+      } catch (e) {
+        printWarning('Error while trying to load font family "$family":\n$e');
+        return null;
+      }
+    }
     // try/catch because `new FontFace` can crash with an improper font family.
     try {
       final DomFontFace fontFace = createDomFontFace(family, asset, descriptors);
-      _fontLoadingFutures.add(fontFace.load().then((_) {
-        domDocument.fonts!.add(fontFace);
-      }, onError: (dynamic e) {
-        printWarning('Error while trying to load font family "$family":\n$e');
-      }));
+      _fontLoadingFutures.add(fontFaceLoad(fontFace));
     } catch (e) {
       printWarning('Error while loading font family "$family":\n$e');
     }
   }
 
+  void registerDownloadedFonts() {
+    if (_downloadedFonts.isEmpty) {
+      return;
+    }
+
+    for (final DomFontFace? fontFace in _downloadedFonts) {
```

Comment:
It looks like you can just use `final DomFontFace fontFace` here and get rid of the null check since only non-null fonts are added to `_downloadedFonts`